File: catalog-core/src/main/java/ro/ubb/catalog/core/service/StaffService.java
package ro.ubb.catalog.core.service;

import ro.ubb.catalog.core.model.Staff;

import java.util.List;

public interface StaffService {
    List<Staff> getAllStaff();

    Staff saveStaff(Staff staff);
    Staff updateStaff(Long id, Staff staff);
    void deleteStaff(Long id);

}

File: webapp/src/app/staff/shared/staff.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Staff} from "./staff.model";

@Injectable()
export class StaffService{

  private staffUrl = "http://localhost:8080/api/staff";

  constructor(private httpClient:HttpClient){}

  getStaff(): Observable<Staff[]> {
    return this.httpClient
      .get<Array<Staff>>(this.staffUrl);
  }

  saveStaff(staff: Staff): Observable<Staff> {
    return this.httpClient
      .post<Staff>(this.staffUrl, staff);
  }
  updateStaff(staff: Staff): Observable<Staff> {
    const url = `${this.staffUrl}/${staff.id}`;
    return this.httpClient.put<Staff>(url, staff);
  }

  deleteStaff(id: number): Observable<any> {
    const url = `${this.staffUrl}/${id}`;
    return this.httpClient
      .delete(url);
  }

}
